shared/domains/okTurtles/events/index.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+// =======================
+// Domain: Event publish/subscribe
+// =======================
+
+import sbp from '../../../sbp'
+
+export default {
+  // TODO: add ability to unregister listeners
+  '/on': function (event, handler) {
+    sbp('okTurtles.data/deepAdd', `events/${event}/listeners`, handler)
+  },
+  '/once': function (event, handler) {
+    sbp('okTurtles.data/deepAdd', `events/${event}/listenOnce`, handler)
+  },
+  '/emit': function (event, data) {
+    const listeners = sbp('okTurtles.data/deepGet', `events/${event}/listeners`) || []
+    listeners.forEach(listener => listener({event, data}))
+    // TODO next up in SBP conversion: listener => sbp(listener, event, data)
+    const listenOnce = sbp('okTurtles.data/deepGet', `events/${event}/listenOnce`) || []
+    listenOnce.forEach(listener => listener({event, data}))
+    sbp('okTurtles.data/deepSet', `events/${event}/listenOnce`, [])
```

Comment:
Copying from Slack:

```
greg [7:32 PM]
ok, yeah, it clears it there. but still, none of the `deep*` methods are needed
you could rewrite this to use just `okTurtles.data/add` and it would work just the same
and faster

aniko [7:33 PM]
but anyways, I am still not convinced that we won’t ever need nesting in our data handling. I can of course remove these and do a flat data structure, I just wanted to give it a last try to come up with a compromise and voice my concerns :slightly_smiling_face:

greg [7:33 PM]
there's no compromise if there's no valid justification :stuck_out_tongue:
you can keep the definition/implementation of those methods, but they're not needed to implement `okTurtles.events`, so they shouldn't be used there
maybe they will be come in handy for something, but right now i can't think of thing, and every precious line of JS counts towards the final bundle size
```